src/lib/drawer/clickOutside.ts

```diff
@@ -0,0 +1,19 @@
+export function clickOutside(node: HTMLElement, ignore?: string) {
+	const handleClick = (event: Event) => {
+		const target = event.target as HTMLElement;
+		if (!event.target || (ignore && target.closest(ignore))) {
+			return;
+		}
+		if (node && !node.contains(target) && !event.defaultPrevented) {
+			node.dispatchEvent(new CustomEvent('click_outside'));
```

Comment:
i'd say instead of using a custom event here, instead pass in a the click handler

e.g.
```ts
function clickOutside(node: HTMLElement, onClick: () => void)
```
```ts
if (node && !node.contains(target) && !event.defaultPrevented) {
  onClick()
}
```
```svelte
use:clickOutside={handleClickOutside}
```